### PR TITLE
Add WS Correction

### PR DESCRIPTION
**Pull request by as2388**
Adapted from ADSS, but with many modifications made to accommodate the new coding_configurations structure of pipeline_configuration.py

### CODE REVIEW

File: src/apply_manual_codes.py [hunk 1]
```diff
@@ -7,10 +7,40 @@
 from core_data_modules.traced_data.io import TracedDataCodaV2IO
 
 from src.lib import PipelineConfiguration
+from src.lib.code_schemes import CodeSchemes
 from src.lib.pipeline_configuration import CodingModes
 
 
 class ApplyManualCodes(object):
+    @staticmethod
+    def _impute_coding_error_codes(user, data):
```
Comment (lukechurch) [inline src/apply_manual_codes.py:16]: @as2388 I don't think I follow how this is related to the WS correction work?

**Review by lukechurch — APPROVED** [review]
LGTM - the ws_correction file is an import from another project, right, rather than freshly developed?

**Comment by as2388** [comment]
> LGTM - the ws_correction file is an import from another project, right, rather than freshly developed?

ws_correction is adapted to use coding_configurations instead of code_scheme/binary_code_scheme fields

File: src/apply_manual_codes.py (hunk 1 above)
Comment (as2388) [reply src/apply_manual_codes.py:16]: ws_correction generates CE codes, which then need to be applied to all code schemes.

**Pull request merged**